mate-panel/panel-addto.c

```diff
@@ -517,16 +517,20 @@ panel_addto_prepend_directory (GSList             **parent_list,
 			       const char          *filename)
 {
 	PanelAddtoAppList *data;
+	GIcon              *gicon;
 
 	data = g_new0 (PanelAddtoAppList, 1);
+	gicon = matemenu_tree_directory_get_icon (directory);
 
 	data->item_info.type          = PANEL_ADDTO_MENU;
 	data->item_info.name          = g_strdup (matemenu_tree_directory_get_name (directory));
 	data->item_info.description   = g_strdup (matemenu_tree_directory_get_comment (directory));
-	data->item_info.icon          = g_strdup (matemenu_tree_directory_get_icon (directory));
+	data->item_info.icon          = g_icon_to_string(gicon);
 	data->item_info.menu_filename = g_strdup (filename);
 	data->item_info.menu_path     = matemenu_tree_directory_make_path (directory, NULL);
 	data->item_info.static_data   = FALSE;
+	if (gicon != NULL)
+		g_object_unref(gicon);
```

Comment:
Do we need this unref? It's not done in other places, and `matemenu_tree_directory_get_icon` has "transfer none" in the description.

Reply:
Yes, gicon created by matemenu_tree_directory_get_icon dose not need unref.